### DISCUSSION TITLE
Students repo locked after deadline

### DISCUSSION THREAD

**Discussion by augustusbuckman** [community/community#64733] [GitHub Education]
### Hi everyone,

I created an assignment on git classroom and set auto-grading functionality.
But after the deadline for submission expires, the students cannot push to the repo even after the submission date is extended

**Comment by arafajs** [comment]
When you extend the deadline for an assignment, GitHub Classroom does not automatically update the deadlines for the autograding tests. This means that students will still be unable to push to the repo after the original deadline has passed.

To fix this, you need to manually update the deadlines for the autograding tests. To do this, follow these steps:

- Go to the assignment in GitHub Classroom.
- Click on the "Autograding" tab.
- Click on the "Edit" button next to the test that you want to update.
- In the "Deadline" field, enter the new deadline for the test.
- Click on the "Save" button.

You need to repeat these steps for each autograding test that you want to update.

Once you have updated the deadlines for the autograding tests, students should be able to push to the repo even after the original deadline has passed.

Here are some additional things to keep in mind:

1. The new deadline cannot be in the past.
2. If you have already graded any student submissions, you will need to regrade them after you update the deadlines for the autograding tests.

**Reply by rafaelcorsi** [reply]
This does not work if the deadline is set to 'This is a cutoff date', then once the assignments reach this deadline, GitHub changes the students' permissions on the repo to 'read-only'. However, if the deadline is subsequently edited, GitHub doesn't revert the permissions back to 'write'.

**Reply by lief-y** [reply]
I had the same issue. Any suggestion how to resolve it? Thank you!

**Reply by rafaelcorsi** [reply]
yes, on the next assignment do not use this resource .. this is my solution

**Comment by cfcurtis** [comment]
I'm encountering a similar issue - I extended the deadline for a single student, and while the student was able to push changes, it doesn't get autograded. I don't want to edit the assignment to extend the deadline for the whole class, and there doesn't seem to be a way to do that for one student.

For now I'm manually adding an extra point to that student's lab work, but it would be nice if the "extension" mechanism behaved as expected.
